src/systems/system.C

```diff
@@ -733,13 +733,64 @@ NumericVector<Number> & System::add_vector (std::string_view vec_name,
   // Return the vector if it is already there.
   auto it = this->_vectors.find(vec_name);
   if (it != this->_vectors.end())
-    return *it->second;
+    {
+      // If the projection setting has *upgraded*, change it.
```

Comment:
We discussed marking `System::add_vector` parallel_only, right?

Reply:
Yeah; sorry it slipped my mind.

Looking through, there's quite a few methods of similar type that ought to be marked parallel_only(); I'll make a separate PR for the lot.